infra/src/main/kotlin/fi/fta/geoviite/infra/ratko/RatkoLocationTrackService.kt

```diff
@@ -155,13 +175,10 @@ class RatkoLocationTrackService @Autowired constructor(
         requireNotNull(layoutLocationTrack.externalId) {
             "Cannot update location track metadata without location track oid, id=${layoutLocationTrack.id}"
         }
-        requireNotNull(layoutLocationTrack.alignmentVersion) {
-            "Location track is missing geometry, id=${layoutLocationTrack.id}"
-        }
 
         val geocodingContext = geocodingService.getGeocodingContextAtMoment(layoutLocationTrack.trackNumberId, moment)
 
-        alignmentDao.fetchMetadata(layoutLocationTrack.alignmentVersion)
+        alignmentDao.fetchMetadata(layoutLocationTrack.getAlignmentVersionOrThrow())
```

Comment:
Olen siellätäällä korvannut myös näitä: `locationTrack.alignmentVersion` -> `locationTrack.getAlignmentVersionOrThrow()`. Tuo on siis ollut jo vaihtoehto hyvän aikaa, mutta ei alusta alkaen, joten monissa paikoin koodipohjaa edelleen käsitellään turhaan nulleja.

Mistä noissa on siis kyse: kannassa tuo on non-null column, eli kun se olio on tallennettu/haettu kannasta, sillä on oltava alignmentversio. Itse kenttä on null vain sitä varten että olio voidaan luoda ennen tallennusta. Tästä johtuen on OK käyttää tuota non-nullia suoraan tuolla metodilla kun tiedetään että kyseessä on kannasta löytyvä rivi. Tämän voisi ideaalimaailmassa ratkaista paremminkin erillisillä tyypeillä mutta se on homma toiselle päivälle.